wordpress-mobile/WordPress-iOS
# Media: Legacy support for MediaService implementation.

This PR adds legacy support for the original `MediaService` implementation. This also adds support for existing `Media` objects that were not exported with the new file and URL specifications.

Why legacy support? Moving forward, I don't want to just merge in use of the new exporting code (and walk away). Instead, we'll support the original implementation now seen in this category and revert to it via a feature flag. While we bring in use of the new exporting code, we'll elevate it's use through the feature flag support, e.g. internal, testflight, release.

Eventually we can remove this category altogether, only using the new exporting code itself.

Changes:

- Excise the relevant code from `MediaService` and support it via `MediaService+Legacy`.
- Update methods just a bit.
- Support existing Media objects and their URLs by using the relevant `MediaFileManager`.
- Write new Media objects, with the old code, to the expected `MediaFileManager` cache URLs.

To test:
1. Build and run.
2. Check that the Media Library in-app loads as expected.
3. Check that adding Media to a post in the editor and publishing works as expected.
4. Run all tests.

Needs review: @frosty, @SergioEstevao, nearing the end here folks! 😅
